datasetUtils/createJob.go

```diff
@@ -38,34 +38,56 @@ func CreateArchivalJob(client *http.Client, APIServer string, user map[string]st
 		Files []string `json:"files"`
 	}
 
-	type jobparamsStruct struct {
-		TapeCopies string `json:"tapeCopies"`
-		Username   string `json:"username"`
+	type jobParamsStruct struct {
+		TapeCopies  string          `json:"tapeCopies"`
+		Username    string          `json:"username"`
+		DatasetList []datasetStruct `json:"datasetList"`
+	}
+
+	type createJobDto struct {
+		JobType      string          `json:"type"`
+		JobParams    jobParamsStruct `json:"jobParams"`
+		OwnerUser    string          `json:"ownerUser"`
+		OwnerGroup   string          `json:"ownerGroup"`
+		ContactEmail string          `json:"contactEmail"`
+	}
+
+	if len(accessGroups) <= 0 {
+		return "", fmt.Errorf("no access groups associated with user")
 	}
 
-	jobMap := make(map[string]interface{})
-	jobMap["emailJobInitiator"] = user["mail"]
-	jobMap["type"] = "archive"
 	//jobMap["creationTime"] = time.Now().Format(time.RFC3339)
 	// TODO these job parameters may become obsolete
 	tc := "one"
 	if *tapecopies == 2 {
 		tc = "two"
 	}
-	jobMap["jobParams"] = jobparamsStruct{tc, user["username"]}
-	jobMap["jobStatusMessage"] = "jobSubmitted"
 
-	emptyfiles := make([]string, 0)
+	emptyfiles := []string{}
+	dsMap := make([]datasetStruct, len(datasetList))
+	for i, dataset := range datasetList {
+		dsMap[i] = datasetStruct{dataset, emptyfiles}
+	}
 
-	var dsMap []datasetStruct
-	for i := 0; i < len(datasetList); i++ {
-		dsMap = append(dsMap, datasetStruct{datasetList[i], emptyfiles})
+	// TODO how the heck can I add a dataset list with the new format?
+	// jobMap["datasetList"] = dsMap
+	createJob := createJobDto{
+		JobType: "archive",
+		JobParams: jobParamsStruct{
+			TapeCopies:  tc,
+			Username:    user["username"],
+			DatasetList: dsMap,
+		},
+		OwnerUser:    user["username"],
+		OwnerGroup:   accessGroups[0],
```

Comment:
Actually, the datasetArchiver command needs to be changed as well to require the user to input a desired ownerGroup as otherwise there's no way to know which access group the user wants to use. Probably the reason I implemented it like this in the first place.

Reply:
The job batching process is also going to be totally screwed up by this as before I think the backend just used the ownerGroup of the dataset without requiring the user to specify it.

Reply:
The above issue is only affected by one command, as others are definitely only working with a single owner group. I fixed it anyways in the latest commit.

Reply:
I think what you pointed out is indeed an issue we should raise with Spencer, and maybe discuss why the new BE requires owners on the jobs. Maybe only the person who triggered the job should be able to see it?

Reply:
if it's only the person themself that needs see it then the backend should already know who's making the request. If it's supposed to be the same as the dataset, why doesn't the backend just get the ownerGroups from the affected datasets, check against the user's accessGroups and set a list of groups who can see this job? 

I don't know the answers because I'm not working on the jobs system, and we shouldn't probably be hung up on it as the v4 job system is not merged yet anyways